snyk/run.sh

```diff
@@ -37,20 +37,22 @@ scan_golang () {
 rm -rf "$DIR"/results
 mkdir -p "$DIR"/results
 
-scan_golang /repository/inbm/trtl trtl  >"$DIR"/results/snyk-trtl.html
-scan_golang /repository/inbm/fpm/inb-provision-cloud inb-provision-cloud >"$DIR"/results/snyk-inb-provision-cloud.html
-scan_golang /repository/inbm/fpm/inb-provision-certs inb-provision-certs >"$DIR"/results/snyk-inb-provision-certs.html
+fail=0
+scan_golang /repository/inbm/trtl trtl  >"$DIR"/results/snyk-trtl.html || fail=1
+scan_golang /repository/inbm/fpm/inb-provision-cloud inb-provision-cloud >"$DIR"/results/snyk-inb-provision-cloud.html || fail=1
+scan_golang /repository/inbm/fpm/inb-provision-certs inb-provision-certs >"$DIR"/results/snyk-inb-provision-certs.html || fail=1
 
```

Comment:
what does the fail=1 do?

Reply:
@nmgaston instead of failing the whole script when the scan finds vulnerabilities, we set a local variable fail=1 and then check it at the very end of all the scans, so we can exit with code 1 and turn teamcity red if any of the individual scans. But we still get full results in the artifacts.